crates/rustc_codegen_spirv/src/codegen_cx/constant.rs

```diff
@@ -374,23 +386,88 @@ impl<'tcx> CodegenCx<'tcx> {
                 .tcx
                 .sess
                 .fatal("Cannot create const alloc of type void"),
-            SpirvType::Bool => {
-                self.constant_bool(DUMMY_SP, self.read_alloc_val(alloc, offset, 1) != 0)
-            }
-            SpirvType::Integer(width, _) => {
-                let v = self.read_alloc_val(alloc, offset, (width / 8) as usize);
-                self.constant_int(ty, v as u64)
-            }
-            SpirvType::Float(width) => {
-                let v = self.read_alloc_val(alloc, offset, (width / 8) as usize);
-                match width {
-                    32 => self.constant_f32(DUMMY_SP, f32::from_bits(v as u32)),
-                    64 => self.constant_f64(DUMMY_SP, f64::from_bits(v as u64)),
-                    other => self
+            SpirvType::Bool
+            | SpirvType::Integer(..)
+            | SpirvType::Float(_)
+            | SpirvType::Pointer { .. } => {
+                let (primitive, size) = match ty_concrete {
```

Comment:
why not `let size = ty_concrete.sizeof(self);`? would save quite a few lines of code and duplication